crates/services/src/service.rs

```diff
@@ -83,6 +83,47 @@ pub trait RunnableService: Send {
     ) -> anyhow::Result<Self::Task>;
 }
 
+/// The result of a single iteration of the service task
+pub enum TaskRunResult {
+    /// Request the task to be run again
+    Continue,
+    /// Request the task to be abandoned
+    Stop,
+    /// Request the task to be run again, but report an error
+    ErrorContinue(anyhow::Error),
```

Comment:
do you envision to have an `ErrorStop` variant in the future?

Reply:
We don't have that use case currently. Would make sense wouldn't it? But YAGNI for now.

Reply:
was just curious :) 
Yeah, I don't think it makes sense to add it now if you don't have a use for it